Add hydrogen 0.9.7-rc1

miccal opened the pull request:
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked that the cask was not already refused in [closed issues].

Closes [#26865](https://github.com/caskroom/homebrew-cask/issues/26865).

fanquake commented:
```
audit for hydrogen: failed
 - download not possible: sha256 mismatch
Expected: 606a33ad624beecc849f8d651b267ee9cea28dbf7e3aea951b9ad4b48c73b0b3
Actual: 86eb2a3cefd7e07ad41d1332b681a75a9ea052cb96910dec0cc4429b65a9c7a8
File: /Users/travis/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
```

miccal commented:
This has got me confused - running `brew cask fetch https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb` results in a file with a different sha256 every time?
```bash
-bash-4.4.5- /Users/miccal [> brew cask fetch https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading external files for Cask https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
######################################################################## 100.0%
==> Downloading https://downloads.sourceforge.net/projects/hydrogen/files/Hydrogen/0.9.7%20Binaries/hydrogen-0.9.7-rc1.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask hydrogen
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 606a33ad624beecc849f8d651b267ee9cea28dbf7e3aea951b9ad4b48c73b0b3
\[\e[1;38;5;2m\] Actual: cdb5de0a182d49f2e0be28203e62a42debf7a0035fe9f55d27b960ff17d3cf8f
File: /Users/miccal/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
To retry an incomplete download, remove the file above.
-bash-4.4.5- /Users/miccal [> brew cask cleanup
==> Removing cached downloads
/Users/miccal/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
/Users/miccal/Library/Caches/Homebrew/Cask/hydrogen.rb
==> This operation has freed approximately 43K of disk space.
-bash-4.4.5- /Users/miccal [> brew cask fetch https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading external files for Cask https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
######################################################################## 100.0%
==> Downloading https://downloads.sourceforge.net/projects/hydrogen/files/Hydrogen/0.9.7%20Binaries/hydrogen-0.9.7-rc1.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask hydrogen
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 606a33ad624beecc849f8d651b267ee9cea28dbf7e3aea951b9ad4b48c73b0b3
Actual: 52fdb8412329037449773b93968a036b7fb789e7906028828b5cd6271c6234ee
File: /Users/miccal/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
To retry an incomplete download, remove the file above.
-bash-4.4.5- /Users/miccal [> brew cask cleanup
==> Removing cached downloads
/Users/miccal/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
/Users/miccal/Library/Caches/Homebrew/Cask/hydrogen.rb
==> This operation has freed approximately 43K of disk space.
-bash-4.4.5- /Users/miccal [> brew cask fetch https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading external files for Cask https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
==> Downloading https://raw.githubusercontent.com/miccal/homebrew-cask/e078eb9d8aa2a9a9f0c7ff84774aca5dd81b956c/Casks/hydrogen.rb
######################################################################## 100.0%
==> Downloading https://downloads.sourceforge.net/projects/hydrogen/files/Hydrogen/0.9.7%20Binaries/hydrogen-0.9.7-rc1.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask hydrogen
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 606a33ad624beecc849f8d651b267ee9cea28dbf7e3aea951b9ad4b48c73b0b3
Actual: 27688040de26b906d85b454b788cbe78c5177c5fffdefe49e22c556d590d81b6
File: /Users/miccal/Library/Caches/Homebrew/Cask/hydrogen--0.9.7,rc1.dmg
To retry an incomplete download, remove the file above.
```

fanquake commented:
Not sure what's happening there, but I just tried downloading it three times from this link `https://sourceforge.net/projects/hydrogen/files/Hydrogen/0.9.7%20Binaries/hydrogen-0.9.7-rc1.dmg/download` and seem to be getting the same shasum each time.

```
daeaa3e7a6fc5ee61a4c59116c90104d2393fd71c8782f136b48bed58d6082db  hydrogen-0.9.7-rc1.dmg
daeaa3e7a6fc5ee61a4c59116c90104d2393fd71c8782f136b48bed58d6082db  hydrogen-0.9.7-rc1.dmg
daeaa3e7a6fc5ee61a4c59116c90104d2393fd71c8782f136b48bed58d6082db  hydrogen-0.9.7-rc1.dmg
```

miccal commented:
OK, I changed the URL, but now `brew cask audit` will complain that the sourceforge URL is [not in the right form](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls), which is why I changed it in the first place.

miccal commented:
```
audit for hydrogen: warning
 - SourceForge URL format incorrect. See https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls
```

vitorgalvao commented on Casks/hydrogen.rb:
```diff
@@ -0,0 +1,11 @@
+  version '0.9.7,rc1'
+  sha256 'daeaa3e7a6fc5ee61a4c59116c90104d2393fd71c8782f136b48bed58d6082db'
+
+  # sourceforge.net/projects/hydrogen was verified as official when first introduced to the cask
```
`sourceforge.net/projects/` isn’t enough to uniquely identify the project, so I changed it to `sourceforge.net/projects/hydrogen`.

miccal replied:
Thanks.

vitorgalvao commented on Casks/hydrogen.rb:
```diff
@@ -0,0 +1,11 @@
+  sha256 'daeaa3e7a6fc5ee61a4c59116c90104d2393fd71c8782f136b48bed58d6082db'
+
+  # sourceforge.net/projects/hydrogen was verified as official when first introduced to the cask
+  url "https://sourceforge.net/projects/hydrogen/files/Hydrogen/#{version.before_comma}%20Binaries/hydrogen-#{version.before_comma}-#{version.after_comma}.dmg/download"
```
Have you tried an appcast for this? We have instructions for sourceforge apps as well.

vitorgalvao replied:
> How do you determine the appcast correctly for Sourceforge projects?

You don’t (reliably), you have to try it.

What the problem with `https://sourceforge.net/projects/hydrogen/rss?path=/Hydrogen`? That one seems to work fine, for me.

vitorgalvao replied:
For the URL, make it simply `https://downloads.sourceforge.net/hydrogen/hydrogen-#{version}.dmg`.

vitorgalvao commented on Casks/hydrogen.rb:
```diff
@@ -0,0 +1,13 @@
+cask 'hydrogen' do
+  version '0.9.7,rc1'
```
It can be `0.9.7-rc1`, after the change to `url`.

miccal commented:
All done.

Very strange, when I first created the cask I am certain I followed the advice in [url.md](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls) and initially had the url as `https://downloads.sourceforge.net/{{project_name}}/{{filename}}.{{ext}}`, but it would not download when I ran `brew cask audit --download`.

Sorry for all the trouble.

vitorgalvao commented:
> Sorry for all the trouble.

No trouble at all. All your contributions are appreciated.
